tests/test_device.py

```diff
@@ -694,3 +708,129 @@ def test_outdated_API(self, monkeypatch):
             m.setattr(qml, "version", lambda: "0.0.1")
             with pytest.raises(DeviceError, match="plugin requires PennyLane versions"):
                 qml.device("default.qubit", wires=0)
+
+
+class TestBatchExecution:
+    """Tests for the batch_execute method."""
+
+    def test_results_no_interface(self, tol):
+        """Tests that the correct results are computed using tapes without an interface."""
+        # skip test if interface cannot be imported
+
+        qml.enable_tape()
+
+        a = 0.1
+        b = 0.2
+        tapes = (tp1, tp2)
+
+        dev = qml.device('default.qubit', wires=2)
+        res = dev.batch_execute(tapes)
+
+        exp1 = [np.cos(a), -np.cos(a) * np.sin(b)]
+        exp2 = np.cos(a)
+        expected = [exp1, exp2]
+
+        # need to compare element-wise because np.allclose
+        # does not work for a mix of lists and tensors
+        for r, e in zip(res, expected):
+            assert np.allclose(r, e, atol=tol, rtol=0)
+        assert isinstance(res, list)
+
+        qml.disable_tape()
+
+    def test_results_autograd_interface(self, tol):
+        """Tests that the correct results are computed using tapes with the autograd interface."""
+
+        qml.enable_tape()
```

Comment:
I'm not sure the interface tests need this either, since the test explicitly uses tapes (not the qnode decorator)